arrows/core/metadata_map_io_csv.cxx

```diff
@@ -332,10 +350,37 @@ ::save_( std::ostream& fout,
 
   fout << std::endl;
 
+  uint64_t next_timestamp = d_->every_n_microseconds;
+  uint64_t next_frame = 1;
   for( auto const& frame_data : data->metadata() )
   {
     for( auto const& metadata_packet : frame_data.second )
     {
+      // Write only at the specified frequency
+      auto const timestamp = metadata_packet->timestamp();
+      if( d_->every_n_microseconds )
+      {
+        if( !timestamp.has_valid_time() ||
+            timestamp.get_time_usec() < next_timestamp )
+        {
+          continue;
+        }
+        next_timestamp +=
+          ( ( timestamp.get_time_usec() - next_timestamp ) /
+            d_->every_n_microseconds + 1 ) * d_->every_n_microseconds;
+      }
+      if( d_->every_n_frames )
```

Comment:
What happens to the export frequency if both every_n_microseconds and every_n_frames are set? Does there need to be a check added somewhere to make sure one isn't overwriting the other? And if both are set: which one gets used by default - or does it error?

Reply:
The current behavior is that only frames which pass both checks will be exported. So if the frame check limits us to every 5 frames and the microseconds check (effectively) limits us to every 3 frames, only every 5 frames will be exported. However, ideally this should not matter because it will error out once `check_configuration()` fails.

Reply:
Ok, sounds good. Is check_configuration called in a different file? And should it be added in here just to double check that only one of these is set before proceeding?

Reply:
Added.

Reply:
Cool, LGTM now